pytest_rts/utils/testgetter.py

```diff
@@ -0,0 +1,85 @@
+"""Module for TestGetter class"""
+
+
+class TestGetter:
+    """Class to query tests and information from the mapping database"""
+
+    def __init__(self, connection):
+        """Set the connection"""
+        self.connection = connection
+
+    @property
+    def existing_tests(self):
+        """Test function names that exist in the mapping database"""
+        return {
+            x[0]
+            for x in self.connection.execute(
+                "SELECT context FROM test_function"
+            ).fetchall()
+        }
+
+    @property
+    def newly_added_tests(self):
+        """New tests collected but not yet mapped"""
+        return {
+            x[0]
+            for x in self.connection.execute("SELECT context FROM new_tests").fetchall()
+        }
+
+    @property
+    def test_function_runtimes(self):
+        """Dictionary with saved runtimes for each test function"""
+        return {
+            x[0]: x[1]
+            for x in self.connection.execute(
+                "SELECT context, duration FROM test_function"
+            ).fetchall()
+        }
+
+    def get_tests_from_srcfiles(self, changed_lines, file_id):
+        """Query tests from source code files based on changed line numbers"""
+        tests = []
+        for line_number in changed_lines:
+            data = self.connection.execute(
+                """ SELECT DISTINCT context
+                    FROM test_function
+                    JOIN test_map ON test_function.id == test_map.test_function_id
+                    WHERE test_map.file_id = ?
+                    AND test_map.line_id = ?  """,
```

Comment:
would it be faster to use something like `f"test_map.line_id IN {changed_lines}"` and avoid `for` loop